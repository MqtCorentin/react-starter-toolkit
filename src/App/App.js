import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { OidcProvider, useOidcUser, useOidcAccessToken, OidcSecure } from '@axa-fr/react-oidc-context';
import Routes from 'App/Routes';
import UserProvider from 'App/UserProvider';
import FetchProvider from 'App/FetchProvider';
import NotificationProvider from 'App/NotificationProvider';
import { PropTypes } from 'prop-types';

const App = ({ oidc, fetchConfig, apiUrl, baseUrl, OidcProviderCmpt, OidcSecureCmpt, useOidcUserFn, useOidcAccessTokenFn }) => (
  <OidcProviderCmpt configuration={oidc}>
    <UserProvider useOidcUser={useOidcUserFn}>
      <OidcSecureCmpt>
        <FetchProvider apiUrl={apiUrl} fetchConfig={fetchConfig} useOidcAccessToken={useOidcAccessTokenFn}>
          <NotificationProvider>
            <Router basename={baseUrl}>
              <Routes />
            </Router>
          </NotificationProvider>
        </FetchProvider>
      </OidcSecureCmpt>
    </UserProvider>
  </OidcProviderCmpt>
);

App.propTypes = {
  OidcProviderCmpt: PropTypes.func,
  OidcSecureCmpt: PropTypes.func,
  useOidcUserFn: PropTypes.func,
  useOidcAccessTokenFn: PropTypes.func,
};

App.defaultProps = {
  OidcProviderCmpt: OidcProvider,
  OidcSecureCmpt: OidcSecure,
  useOidcUserFn: useOidcUser,
  useOidcAccessTokenFn: useOidcAccessToken,
};

export default App;
