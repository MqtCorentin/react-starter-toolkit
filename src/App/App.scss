@import '@axa-fr/react-toolkit-form-core/dist/form.scss';
@import '@axa-fr/react-toolkit-title/dist/title.scss';
@import '@axa-fr/react-toolkit-button/dist/button.scss';
@import '@axa-fr/react-toolkit-form-input-text/dist/inputtext.scss';
@import '@axa-fr/react-toolkit-form-input-select/dist/select.scss';
@import '@axa-fr/react-toolkit-form-input-radio/dist/radio.scss';
@import '@axa-fr/react-toolkit-form-input-checkbox/dist/checkbox.scss';
@import '@axa-fr/react-toolkit-form-input-file/dist/file.scss';
@import '@axa-fr/react-toolkit-form-input-textarea/dist/textarea.scss';
@import '@axa-fr/react-toolkit-loader/dist/spinner.scss';
@import '@axa-fr/react-toolkit-popover/dist/popover.scss';
@import '@axa-fr/react-toolkit-link/dist/link.scss';
@import '@axa-fr/react-toolkit-modal-default/dist/modal.scss';
@import '@axa-fr/react-toolkit-table/dist/table.scss';
@import '@axa-fr/react-toolkit-alert/dist/alert.scss';
@import '@axa-fr/react-toolkit-badge/dist/badge.scss';
@import '../shared/scss/fixes-toolkit.scss';
