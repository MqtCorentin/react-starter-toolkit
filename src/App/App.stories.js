import React from 'react';
import { storiesOf } from '@storybook/react';
import notes from 'App/App.md';

storiesOf('Get Started', module).add('Informations', () => <h1>Projet OASIS</h1>, {
  notes: { markdown: notes },
});
