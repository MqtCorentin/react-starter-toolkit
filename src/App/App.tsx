import { BrowserRouter as Router } from 'react-router-dom';
import { OidcProvider, useOidcUser, useOidcAccessToken, OidcSecure } from '@axa-fr/react-oidc-context';
import Routes from 'App/Routes';
import UserProvider from 'App/UserProvider';
import FetchProvider from 'App/FetchProvider';
import NotificationProvider from 'App/NotificationProvider';
import type { TEnvironment } from './EnvironmentProvider';

type TApp = TEnvironment & {
  OidcProviderCmpt?: typeof OidcProvider;
  OidcSecureCmpt?: typeof OidcSecure;
  FetchProviderCmpt?: typeof OidcSecure;
  useOidcUserFn?: typeof useOidcUser;
  useOidcAccessTokenFn?: typeof useOidcAccessToken;
};

const App = ({
  oidc,
  fetchConfig,
  apiUrl,
  baseUrl,
  OidcProviderCmpt = OidcProvider,
  OidcSecureCmpt = OidcSecure,
  FetchProviderCmpt = FetchProvider,
  useOidcAccessTokenFn = useOidcAccessToken,
  useOidcUserFn = useOidcUser,
}: TApp) => (
  <OidcProviderCmpt configuration={oidc}>
    <UserProvider useOidcUser={useOidcUserFn}>
      <OidcSecureCmpt>
        <FetchProviderCmpt apiUrl={apiUrl} fetchConfig={fetchConfig} useOidcAccessTokenFn={useOidcAccessTokenFn}>
          <NotificationProvider>
            <Router basename={baseUrl}>
              <Routes />
            </Router>
          </NotificationProvider>
        </FetchProviderCmpt>
      </OidcSecureCmpt>
    </UserProvider>
  </OidcProviderCmpt>
);

export default App;
