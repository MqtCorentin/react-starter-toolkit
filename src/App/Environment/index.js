export { default, withEnvironment, EnvironmentProvider, useEnv } from './Environment';
