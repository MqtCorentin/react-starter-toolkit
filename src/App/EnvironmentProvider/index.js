export { default, EnvironmentContext, useEnv } from './Environment';
