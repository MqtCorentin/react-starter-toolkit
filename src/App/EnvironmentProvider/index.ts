export { default, EnvironmentContext, useEnv } from './Environment';
export type { TEnvironment, TEnvironmentState } from './Environment';
