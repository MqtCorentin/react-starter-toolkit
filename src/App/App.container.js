import React, { useContext } from 'react';
import { Loader } from '@axa-fr/react-toolkit-all';
import EnvironmentProvider, { EnvironmentContext } from 'App/EnvironmentProvider';
import App from './App';

export const AppContainer = ({ EnvironmentContextObj = EnvironmentContext }) => {
  const { environment } = useContext(EnvironmentContextObj);
  return !environment ? <Loader mode="get" className="af-loader af-loader--fullscreen" /> : <App {...environment} />;
};

const AppWithEnvironment = () => (
  <EnvironmentProvider>
    <AppContainer />
  </EnvironmentProvider>
);

export default AppWithEnvironment;
