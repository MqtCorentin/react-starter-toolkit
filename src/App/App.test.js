import React, { createContext } from 'react';
import { WrapperStaticRouter } from 'shared/testsUtils';
import { render, act, waitFor } from '@testing-library/react';
import App from './App';

const useOidcUserMock = jest.fn().mockReturnValue({
  oidcUser: {
    name: 'Samuel Gomez',
  },
});
const useOidcAccessTokenMock = jest.fn().mockReturnValue({
  accessToken: 'accessTokenfdsfdsqgvqvsqfs',
});

const envMock = {
  baseUrl: '/',
  OidcProviderCmpt: ({ children }) => <>{children}</>,
  OidcSecureCmpt: ({ children }) => <>{children}</>,
  useOidcUserFn: useOidcUserMock,
  useOidcAccessTokenFn: useOidcAccessTokenMock,
};

it('Should render App', () => {
  const { asFragment } = render(<App {...envMock} />, { wrapper: WrapperStaticRouter });
  expect(asFragment()).toMatchSnapshot();
});
