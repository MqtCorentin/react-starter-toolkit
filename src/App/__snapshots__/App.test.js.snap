// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders RouteBase component without crashing 1`] = `
<DocumentFragment>
  <div
    style="min-height: 94px; position: relative;"
  >
     
  </div>
  <div
    class="af-nav-container"
  >
    <button
      aria-controls="mainmenu"
      aria-haspopup="true"
      class="mask fade "
      id="togglemenu"
      type="button"
    />
    <nav
      aria-label="Menu principal"
      class="af-nav af-drawer left "
      role="navigation"
    >
      <div
        class="container af-nav__wrapper"
      >
        <div
          class="af-nav__aside-header"
        >
          <h4
            class="af-nav__aside-title"
          >
            Menu
          </h4>
          <button
            aria-controls="mainmenu"
            aria-haspopup="true"
            class="af-nav__aside-close"
            id="togglemenu"
            type="button"
          >
            ×
          </button>
        </div>
        <ul
          aria-labelledby="togglemenu"
          class="af-nav__list "
          id="mainmenu"
          role="menubar"
        >
          <li
            class="af-nav__item af-nav__item--active "
            role="none"
          >
            <a
              aria-expanded="false"
              aria-haspopup="true"
              aria-label="Accueil"
              class="af-nav__link"
              href="/"
              role="menuitem"
              tabindex="0"
            >
              Accueil
            </a>
          </li>
          <li
            class="af-nav__item "
            role="none"
          >
            <a
              aria-expanded="false"
              aria-haspopup="true"
              aria-label="Membres"
              class="af-nav__link"
              href="/members"
              role="menuitem"
              tabindex="-1"
            >
              Membres
            </a>
          </li>
          <li
            class="af-nav__item af-nav__item--haschild "
            role="none"
          >
            <a
              aria-expanded="false"
              aria-haspopup="true"
              aria-label="Forms"
              class="af-nav__link"
              href="/form"
              role="menuitem"
              tabindex="-1"
            >
              Forms
            </a>
            <i
              class="glyphicon glyphicon-arrow-xs-down"
            />
            <ul
              aria-label=""
              class="af-nav__list af-nav__list--children"
              role="menu"
            >
              <li
                class="af-nav__item "
                role="none"
              >
                <a
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="Champs classiques"
                  class="af-nav__link"
                  href="/form/classics-fields"
                  role="menuitem"
                  tabindex="-1"
                >
                  Champs classiques
                </a>
              </li>
              <li
                class="af-nav__item "
                role="none"
              >
                <a
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="Champ datepicker"
                  class="af-nav__link"
                  href="/form/datepicker"
                  role="menuitem"
                  tabindex="-1"
                >
                  Champ datepicker
                </a>
              </li>
              <li
                class="af-nav__item "
                role="none"
              >
                <a
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="Champ slider"
                  class="af-nav__link"
                  href="/form/slider"
                  role="menuitem"
                  tabindex="-1"
                >
                  Champ slider
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <a
    aria-controls="mainmenu"
    aria-haspopup="true"
    class="btn af-title-bar__mobile-menu af-btn--circle"
    href="#"
    id="togglemenu"
    role="button"
    tabindex="0"
    title="Toggle menu"
  >
    <i
      class="glyphicon glyphicon-menu-hamburger"
    />
  </a>
  <div
    class="af-title-bar"
  >
    <div
      class="container af-title-bar__wrapper"
    >
      <h1
        class="af-title-bar__title"
      >
        Accueil
      </h1>
    </div>
  </div>
  <div
    class="af-main container"
  >
    <h1
      class="af-title--content"
    >
      Tableau de bord
    </h1>
  </div>
  <footer
    class="af-footer"
  >
    <div
      class="container-fluid container"
    >
      <a
        class="af-logo"
        href="https://www.axa.fr/"
        target="blank"
        title="Site Axa"
      >
        <img
          alt="Logo Axa"
          class="af-logo__brand"
          src="logo-axa.svg"
        />
      </a>
      <div
        class="af-footer-content"
      >
        © 2020 AXA Tous droits réservés - v0.1.0
      </div>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Should App render without crashing 1`] = `
<DocumentFragment>
  <div
    style="min-height: 94px; position: relative;"
  >
     
  </div>
</DocumentFragment>
`;
